agent/util-scripts/modules/metaclass.py

```diff
@@ -0,0 +1,98 @@
+from pathlib import Path
+import json
+import os
+
+
+class ToolMetadata:
+    def __init__(self, mode, context, logger):
+        self.logger = logger
+        if mode not in ("redis", "json"):
+            raise Exception()
```

Comment:
The Exception() objects could have descriptions; though I'm not sure it's worth subclassing Exception it might be a bit cleaner.